scaling_performance/using_cpu_manager.adoc

```diff
@@ -0,0 +1,203 @@
+https://access.redhat.com/support/offerings/techpreview/.
+endif::[]
+====
+
+CPU Manager manages groups of CPUs and constrains workloads to specific CPUs.
+
+CPU Manager is useful for workloads that have some of these attributes:
+
+* Require as much CPU time as possible.
+* Are sensitive to processor cache misses.
+* Are low-latency network applications.
+* Coordinate with other processes and benefit from sharing a single processor
+cache.
+
+[[setting-up-cpu-manager]]
+== Setting Up CPU Manager
+
+To set up CPU Manager:
+
+. Optionally, label a node:
++
+----
+# oc label node perf-node.example.com cpumanager=true
+----
+
+. Enable CPU manager support on the target node:
++
+----
+# cat /etc/origin/node/node-config.yaml
+...
+kubeletArguments:
+...
+  feature-gates:
+  - CPUManager=true
+  cpu-manager-policy:
+  - static
+  cpu-manager-reconcile-period:
+  - 5s
+  kube-reserved: <1>
+  - cpu=500m
+
+# systemctl restart atomic-openshift-node
+----
+<1> `kube-reserved` is a required setting. The value may need to be adjusted
+depending on your environment.
+
+. Create a pod that requests a core or multiple cores. Both limits and requests
+must have their CPU value set to a whole integer. That is the number of cores
+that will be dedicated to this pod:
++
+----
+# cat cpumanager.yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  generateName: cpumanager-
+spec:
+  containers:
+  - name: cpumanager
+    image: gcr.io/google_containers/pause-amd64:3.0
+    resources:
+      requests:
+        cpu: 1
+        memory: "1G"
+      limits:
+        cpu: 1
+        memory: "1G"
+  nodeSelector:
+    cpumanager: "true"
+----
+
+. Create the pod:
++
+----
+# oc create -f cpumanager.yaml
+----
+
+. Verify that the pod is scheduled to the node that you labeled:
++
+----
+# oc describe pod cpumanager
+Name:         cpumanager-4gdtn
+Namespace:    test
+Node:         perf-node.example.com/172.31.62.105
+...
+    Limits:
+      cpu:     1
+      memory:  1G
+    Requests:
+      cpu:        1
+      memory:     1G
+...
+QoS Class:       Guaranteed
+Node-Selectors:  cpumanager=true
+                 region=primary
+----
+
+. Verify that the `cgroups` are set up correctly. Get the PID of the pause process:
++
+----
+# systemd-cgls -l
+├─1 /usr/lib/systemd/systemd --system --deserialize 20
+├─kubepods.slice
+│ ├─kubepods-pod0ec1ab8b_e1c4_11e7_bb22_027b30990a24.slice
+│ │ ├─docker-b24e29bc4021064057f941dc5f3538595c317d294f2c8e448b5e61a29c026d1c.scope
+│ │ │ └─44216 /pause
+----
++
+Pods of QoS tier `Guaranteed` are placed within the `kubepods.slice`. Pods of other
+QoS tiers end up in child `cgroups` of `kubepods`.
++
+----
+# cd /sys/fs/cgroup/cpuset/kubepods.slice/kubepods-pod0ec1ab8b_e1c4_11e7_bb22_027b30990a24.slice/docker-b24e29bc4021064057f941dc5f3538595c317d294f2c8e448b5e61a29c026d1c.scope
+# for i in `ls cpuset.cpus tasks` ; do echo -n "$i "; cat $i ; done
+cpuset.cpus 2
+tasks 44216
+----
+
+. Check the task's allowed CPU list:
```

Comment:
IBM Styleguide suggests avoiding possessives.  Maybe " Check the allows CPU list for the task:"